qiskit/pulse/schedule.py

```diff
@@ -1326,44 +1326,33 @@ def filter(
         *filter_funcs: List[Callable],
         channels: Optional[Iterable[Channel]] = None,
         instruction_types: Union[Iterable[abc.ABCMeta], abc.ABCMeta] = None,
-        time_ranges: Optional[Iterable[Tuple[int, int]]] = None,
-        intervals: Optional[Iterable[Interval]] = None,
         check_subroutine: bool = True,
     ):
-        """Return a new ``Schedule`` with only the instructions from this ``ScheduleBlock``
-        which pass though the provided filters; i.e. an instruction will be retained iff
+        """Return a new ``ScheduleBlock`` with only the instructions from this ``ScheduleBlock``
+        which pass though the provided filters; i.e. an instruction will be retained if
         every function in ``filter_funcs`` returns ``True``, the instruction occurs on
-        a channel type contained in ``channels``, the instruction type is contained
-        in ``instruction_types``, and the period over which the instruction operates
-        is *fully* contained in one specified in ``time_ranges`` or ``intervals``.
+        a channel type contained in ``channels``, and the instruction type is contained
+        in ``instruction_types``.
```

Comment:
```suggestion
        in ``instruction_types``.
        
        .. warning::
            Because ``ScheduleBlock`` is not aware of the execution time of 
            the context instructions, filtering out some instructions may
            change the execution time of the remaining instructions.
```

To make it more user friendly :)